src/data/markdown/docs/40 xk6-disruptor/04 Examples/01 Inject Grpc faults into Service.md

```diff
@@ -0,0 +1,443 @@
+
+| Execution       | P95 Req. Duration |  Passed Checks |
+| --------------- | ----------------- | -------------- |
+| Baseline        | 2.09ms            | 100%           |
+| Fault injection | 303.57ms          | 89%            |
+
+
+## Source Code
+
+<Collapsible title="grpc-faults.js">
+
+```javascript
+import { ServiceDisruptor } from 'k6/x/disruptor';
+import grpc from 'k6/net/grpc';
+import { check } from 'k6';
+
+const client = new grpc.Client();
+client.load(['pb'], 'hello.proto');
+
+export default function () {
+    client.connect(
+        __ENV.GRPC_HOST,
+        {
+            plaintext: true,
+            timeout: '1s'
+        }
+    );
+    
+    const data = { greeting: 'Bert' };
+    const response = client.invoke(
+        'hello.HelloService/SayHello',
+        data,
+        {
+            timeout: '1s'
+        }
+    );
+    client.close()
+
+    check(response, {
+        'status is OK': (r) => r && r.status === grpc.StatusOK,
+    });
+
+}
+
+export function disrupt() {
+    if (__ENV.SKIP_FAULTS == "1") {
+        return
+    }
+
+    const disruptor = new ServiceDisruptor('grpcbin','grpcbin')
+
+    // inject errors in requests
+    const fault = {
+        averageDelay: 300,
+        statusCode: grpc.StatusInternal,
+        errorRate: 0.1,
+        port: 9000,
+    }
+    disruptor.injectGrpcFaults(fault, 30)
+}
+
+export const options = {
+    scenarios: {
+        load: {
+            executor: 'constant-arrival-rate',
+            rate: 100,
+            preAllocatedVUs: 10,
+            maxVUs: 100,
+            exec: "default",
+            startTime: '0s',
+            duration: "30s",
+        },
+        disrupt: {
+            executor: 'shared-iterations',
+            iterations: 1,
+            vus: 1,
+            exec: "disrupt",
+            startTime: "0s",
+        }
+    }
+}
+```
+</Collapsible>
+
+
+<Collapsible title="pb/hello.proto">
+
+```
+syntax = "proto2";
+
+package hello;
+
+service HelloService {
+  rpc SayHello(HelloRequest) returns (HelloResponse);
+  rpc LotsOfReplies(HelloRequest) returns (stream HelloResponse);
+  rpc LotsOfGreetings(stream HelloRequest) returns (HelloResponse);
+  rpc BidiHello(stream HelloRequest) returns (stream HelloResponse);
+}
+
+message HelloRequest {
+  optional string greeting = 1;
+}
+
+message HelloResponse {
+  required string reply = 1;
+}
+```
+
+</Collapsible>
+
+## Test setup
+
+The tests requires the deployment of the `grpcbin` application. The application must also be accessible using an external IP available in the `GRPC_HOST` environment variable.
+
+The [manifests](#manifests) below define the resources required for deploying the application and exposing it as a LoadBalancer service.
```

Comment:
```suggestion
The following [manifests](#manifests) define the resources required for deploying the application and exposing it as a LoadBalancer service.
```

The style guidelines say to avoid directional llanguage, like "above" or "below", for accessibility: https://developers.google.com/style/accessibility